src/precice/impl/WatchPoint.cpp

```diff
@@ -99,33 +99,33 @@ void WatchPoint::initialize()
   }
 
   // Find closest edge
-  query::FindClosestEdge findEdge(_point);
-  if (findEdge(*_mesh)) {
-    if (findEdge.getEuclidianDistance() < _shortestDistance) {
-      //         _closestEdge = & findEdge.getClosestEdge ();
+  auto closestEdges = indexTree.getClosestEdges(watchVertex, 1);
```

Comment:
The edge/triangle with the shortest distance may not be an interpolating projection.
Have a look at the implementation in the nearest projection mapping.